From: =?UTF-8?q?S=C3=B8ren=20Andersen?= <san@skov.dk>
Date: Thu, 7 Jul 2016 17:47:32 +0200
Subject: [PATCH] Skov: add support for board variant 20
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add support for board variant 20 (QuadCore, RAM 1G, Flash 4G).

Signed-off-by: Søren Andersen <san@skov.dk>
[Import of b902ce190919 ("Support for board 20 QuadCore Ram 1G, Flash 4G") from "meta-skov-bsp"]
Signed-off-by: Ulrich Ölmann <u.oelmann@pengutronix.de>
---
 arch/arm/boards/skov-dol63x/lowlevel.c | 20 ++++++++++++++++----
 arch/arm/boards/skov-dol63x/version.c  |  6 ++++++
 2 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boards/skov-dol63x/lowlevel.c b/arch/arm/boards/skov-dol63x/lowlevel.c
index 0faba73a1af6..0e0643dfaf22 100644
--- a/arch/arm/boards/skov-dol63x/lowlevel.c
+++ b/arch/arm/boards/skov-dol63x/lowlevel.c
@@ -744,10 +744,12 @@ static inline unsigned skov_dol63x_get_version(int cpu_type)
  * 00001101       C     256 MiB    256 MiB     i.MX6S
  * 00001110       C    1024 MiB    256 MiB     i.MX6Q
  * 00001111       C     512 MiB    256 MiB     i.MX6S
- * 00010000       ?     512 MiB      4 GiB     i.MX6S
- * 00010001       ?    2048 MiB      8 GiB     i.MX6Q
- * 00010010       ?    4096 MiB     16 GiB     i.MX6Q+
- *
+ * 00010000       C     512 MiB      4 GiB     i.MX6S
+ * 00010001       C    2048 MiB      8 GiB     i.MX6Q
+ * 00010010       C    4096 MiB     16 GiB     i.MX6Q+
+ * 00010011       C     512 MiB      2 GiB     i.MX6S
+ * 00010100       C    1024 MiB      4 GiB     i.MX6Q
+ * 
  * This routine does not return if starting the image from SD card or NOR
  * was successful. It restarts skov_dol63x_start() instead
  */
@@ -887,6 +889,16 @@ static void skov_dol63x_init(int cpu_type, unsigned board_variant)
 					&skov_dol63x_calib_2x128Mb_800MHz,
 					&skov_dol63x_cfg_2x128Mb_800MHz);
 		break;
+	case 20: /* i.MX6Q, "Quad_R1G_F2G" */
+		if (cpu_type != IMX6_CPUTYPE_IMX6Q) {
+			pr_err("Invalid SoC! i.MX6Q expected\n");
+			return;
+		}
+		pr_debug("Initializing board variant 20\n");
+		spl_imx6q_dram_init(&skov_dol63x_sysinfo_4x128Mb_1066MHz,
+					&skov_dol63x_calib_4x128Mb_1066MHz,
+					&skov_dol63x_cfg_4x128Mb_1066MHz);
+		break;
 	default:
 		pr_err("Unsupported board revision: 0x%x\n", board_variant);
 		/* FIXME don't continue */
diff --git a/arch/arm/boards/skov-dol63x/version.c b/arch/arm/boards/skov-dol63x/version.c
index d90402056321..8e7423a63a9e 100644
--- a/arch/arm/boards/skov-dol63x/version.c
+++ b/arch/arm/boards/skov-dol63x/version.c
@@ -120,6 +120,12 @@ static const struct board_description dol63x_variants[] = {
 		.soc = "i.MX6S",
 		.dts_compatible = "skov,dol63x-imxdl-revC",
 	},
+	[20] = {
+		.value = "Quad_R1G_F4G",
+		.revision = "C",
+		.soc = "i.MX6Q",
+		.dts_compatible = "skov,dol63x-imxq-revC",
+	},
 };
 
 /*
